src/UserSpaceInstrumentation/include/UserSpaceInstrumentation/InstrumentProcess.h

```diff
@@ -32,12 +32,18 @@ class InstrumentationManager {
 
   [[nodiscard]] static std::unique_ptr<InstrumentationManager> Create();
 
+  struct InstrumentationResult {
+    absl::flat_hash_set<uint64_t> instrumented_function_ids;
+    std::vector<std::string> error_messages;
```

Comment:
If I understand it correctly, the `error_messages` will be associated to the function that couldn't be instrumented only by maybe containing the name of the function. What do you think of using a map of function ids to error messages? This would leave more flexibility where the information is used. For example the client could show the list of problematic functions differently, maybe even allowing specific actions on such functions, and whatnot.